src/common/Exporter.js

```diff
@@ -0,0 +1,80 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define([], factory);
+    } else {
+        root.common_Exporter = factory();
+    }
+}(this, function () {
+    function Exporter(widget) {
+        this._widget = widget;
+    }
+
+    Exporter.prototype.formatData = function (_) {
```

Comment:
That would be fine with me, I originally thought about just doing all of it in Widget.js, but wasn't sure if that would be acceptable. I will update it.